File: src/main/java/demo/bfims/Entities/Inventory/Book.java
package demo.bfims.Entities.Inventory;
import demo.bfims.Enums.ItemType;
import jakarta.persistence.*;

@Entity
@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
public class Book extends PublicationItem {
    public Book(){
        this.setItemType(ItemType.BOOK);
    }

    @Override
    public String toString() {
        return "Book{" +
                "publication=" + publication +
                "} " + super.toString();
    }
}

File: frontend/src/app/DTOs/Inventory/Book.ts
import {PublicationItem} from './PublicationItem';
import {PublicationItemType} from '../../Enums/PublicationItemType';

export class Book extends PublicationItem {

  constructor(...data:any) {
    super(data[0])
    this.publicationItemType = PublicationItemType.BOOK;
  }
}
